spring-batch-infrastructure/src/main/java/org/springframework/batch/item/file/transform/BeanWrapperFieldExtractor.java

```diff
@@ -54,7 +116,12 @@ public Object[] extract(T item) {
 
 		BeanWrapper bw = new BeanWrapperImpl(item);
 		for (String propertyName : this.names) {
-			values.add(bw.getPropertyValue(propertyName));
+			Object myObject = bw.getPropertyValue(propertyName);
+			if (myObject instanceof String && ((String) myObject).contains(delimiter)) {
```

Comment:
Analysing the content of objects and adding quotes is not the responsibility of the `BeanWrapperFieldExtractor`. If we do it here, we should also do it in other `FieldExtractor`s like `RecordFieldExtractor`.

This could be added as a feature by the user in a subclass if needed, like `SmartBeanWrapperFieldExtractor` or `DelimiterAwareBeanWrapperFieldExtractor`.